File: 1598_CrawlerLogFolder/CrawlerLogFolder.cpp

class Solution {
public:
    int minOperations(vector<string>& logs) {
        // own effeciant
        int need = 0;
        for(string s : logs) {
            if(s == "../") {
                if(need == 0) continue;
                need--; 
                continue;
            }
            else if(s == "./") continue;
            else if(s.back() == '/') need++;
        }
        return need;
    }
};

File: 1598_CrawlerLogFolder/CrawlerLogFolder.java

class Solution {
    public int minOperations(String[] logs) {
        // own effeciant
        int need = 0;
        for (String s : logs) {
            if (s.equals("../")) {
                if (need == 0)
                    continue;
                need--;
                continue;
            } else if (s.equals("./"))
                continue;
            else if (s.contains("/"))
                need++;
        }
        return need;
    }
}
